storm-client/src/jvm/org/apache/storm/utils/Utils.java

```diff
@@ -1505,35 +1508,36 @@ public static StormTopology addVersions(StormTopology topology) {
         Map<String, Object> defaultsConf = null;
         Map<String, Object> stormConf = null;
         for (String part: cp) {
-            File f = new File(part);
-            if (f.isDirectory()) {
-                if (defaultsConf == null) {
-                    defaultsConf = readConfIgnoreNotFound(yaml, new File(f, "defaults.yaml"));
-                }
-                
-                if (stormConf == null) {
-                    stormConf = readConfIgnoreNotFound(yaml, new File(f, "storm.yaml"));
+            String wildcardSuffix = File.separator + "*";
```

Comment:
Wouldn't it also work to just check if the part ends with "*"?

Reply:
According to my understanding of Oracle Java classpath doc, it should only expand when file name is '*' rather than like 'abc*'. If 'abc*' is given in file name, it should not expand to abcde.jar or so.

https://docs.oracle.com/javase/8/docs/technotes/tools/unix/classpath.html

Reply:
Thanks, hadn't thought of that. I was thinking of the case where the full path is "*", which I think you've fixed now.